esbuild.js

```diff
@@ -0,0 +1,127 @@
+    background: `${src}/background/background-init.ts`,
+    injected: `${src}/injected/client-init.ts`,
+    popup: `${src}/popup/popup-init.ts`,
+    insights: `${src}/views/insights/initializer.ts`,
+    detailsView: `${src}/DetailsView/details-view-initializer.ts`,
+    devtools: `${src}/Devtools/dev-tool-init.ts`,
+    serviceWorker: `${src}/background/service-worker-init.ts`,
+    debugTools: `${src}/debug-tools/initializer/debug-tools-init.tsx`,
+};
+const devWebExtensionOutdir = path.join(__dirname, 'extension/devBundle');
+const devWebExtensionM3Outdir = path.join(__dirname, 'extension/devMv3Bundle');
+
+const prodWebExtensionOutDir = path.join(__dirname, 'extension/prodBundle');
+
+// Default behavior; builds dev extension.
+let entryFiles = webExtensionEntryFiles;
+let outdir = devWebExtensionOutdir;
+let isProd = false;
+let platform = 'browser';
+let external = [];
+let format = 'esm';
+let define;
+let target = ['chrome90', 'firefox90'];
+let plugins = [CreateStylePlugin()];
+
+switch (argsObj.env) {
+    // Note: currently causes errors when electron app is run.
+    case 'unified':
+        entryFiles = electronEntryFiles;
+        outdir = unifiedOutdir;
+        platform = 'node';
+        external = [];
+        format = 'cjs';
+        break;
+
+    case 'prod':
+        isProd = true;
+        outdir = prodWebExtensionOutDir;
+        delete entryFiles.detailsView;
+        break;
+
+    case 'devM3':
+        outdir = devWebExtensionM3Outdir;
+        define = {
+            global: 'globalThis',
+        };
+        break;
+
+    case 'report':
+        entryFiles = { report: `${src}/reports/package/reporter-factory.ts` };
+        outdir = path.join(__dirname, 'packages/report/bundle');
+
+        // esbuild doesn't have an easy way to ignore node_modules for monorepos,
+        // so this plugin is necessary (marks all node_modules as external).
+        // Thread: https://github.com/evanw/esbuild/issues/619
+        plugins.push(
+            NodeResolve.NodeResolvePlugin({
+                extensions: ['.ts', '.tsx', '.js'],
+                onResolved: resolved => {
+                    if (resolved.includes('node_modules')) {
+                        return {
+                            external: true,
+                        };
+                    }
+                    return resolved;
+                },
+            }),
+        );
+        break;
+}
+
+const config = {
+    plugins,
+    entryPoints: entryFiles,
+    format,
+    platform,
+    external,
+    outbase: src,
+    bundle: true,
+    minify: isProd,
+    sourcemap: !isProd,
+    target,
+    outdir: outdir,
+    outExtension: {
+        '.js': '.bundle.js',
+    },
+    watch: argv.includes('--watch'),
+    logLevel: 'info',
+    define,
+};
+
+esbuild
+    .build(config)
+    .then(() => {
+        if (!isProd) {
+            return;
+        }
+
+        // minification of the details-view bundle leads to a collision with the axe-core package,
```

Comment:
This strategy is going to cause problems for `--watch`, no?

Reply:
It also doubles the size of the production details view bundle, which isn't really thrilling from a user experience perspective

Reply:
On the bundle size portion, I mention that here: https://github.com/microsoft/accessibility-insights-web/issues/5412#issuecomment-1129457320.

It's a difference of 15.7MB to 12.6MB (vs. 13.1MB with webpack). It felt like to me that was _okay_ for now and ideally we can file an issue on esbuild/hopefully get it resolved at some point (since it was due to a naming collision).

Thoughts?

Reply:
Will look into the watch again ... seemed to be working prior fine. Will report back.

Reply:
The issue isn't with the total size of all bundles, but with the size of the details view bundle specifically; just the extra time in the initial load + evaluate of the script is user-perceptible (+80ms) on my relatively powerful dev machine.

I don't think that's a big enough regression to consider it blocking compared to the development cycle time benefits, but it's enough that I think it would be worth investigating if there's maybe a middle ground we could approach. For example, did you already explore whether the axe-core issue still repros with `minify: true, keepNames: true` (gets ~85% of the minification) or with `minifyWhitespace: true, minifySyntax: true, minifyIdentifiers: false` (gets ~60% of the minification)?

Reply:
Gotcha!

Yeah, I did look into those options but wasn't fully satisfied with the result because it seems like a transient issue on my end: currently, minifying everything normally **doesn't** produce this issue. That's without changing anything (i.e. not including the workaround or using the sub-minify variables).

I'm trying to get the error to repro again so I can appropriately test these options but reading through the bundles made it seem like the variable names were still likely to collide ... when the stars align. It's all very odd.

Reply:
I was able to go through my history to find a moment in time that was able to repro the error*. With that, I was able to test a bit more and it seems like minifyIdentifiers set to false leads to a bundle that is workable (I think previously when I tested this, I forgot to unset minify).

\* still don't know what is causing this to happen.